assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * DashboardIdeasWidget component
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Tab from '@material/react-tab';
+import TabBar from '@material/react-tab-bar';
+import { useHash, useMount } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState, useRef, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import whenActive from '../../../../../util/when-active';
+import NewIdeas from './NewIdeas';
+import SavedIdeas from './SavedIdeas';
+import DraftIdeas from './DraftIdeas';
+
+const getHash = ( hash ) => hash ? hash.replace( '#', '' ) : false;
+const isValidHash = ( hash ) => getHash( hash ) in DashboardIdeasWidget.tabToIndex;
+
+const DashboardIdeasWidget = () => {
+	const ideaHubContainer = useRef();
+	const [ hash, setHash ] = useHash();
+	const [ activeTabIndex, setActiveTabIndex ] = useState( DashboardIdeasWidget.tabToIndex[ getHash( hash ) ] );
```

Comment:
The `getHash` function may return `false` if nothing is selected yet. It means that we will use `undefined` as a default value for the `activeTabIndex` state variable. Let's fall back to the first tab index (which is `0`) in this case.

Reply:
Updated @eugene-manuilov